### PR TITLE
add ephemeraljob validating webhook

### LINKED ISSUE #1614: [feature request] add validating webhook for ejob
**What would you like to be added**:

Invalid ejobs should be rejected by validating webhook.

**Why is this needed**:

If invalid ejobs exist in cluster, kruise manager will panic.

### PR DESCRIPTION
**Pull request by ABNER-1**
### Ⅰ. Describe what this PR does
fix #1614 

### Ⅲ. Describe how to verify it
Invalid ejob must be rejected by webhook.
For example:
image

### CODE REVIEW

File: pkg/webhook/ephemeraljob/validating/ephemeraljob_create_update_handler.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		klog.Warningf("Error validate EphemeralJob %s: %v", obj.Name, err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	return admission.ValidationResponse(true, "allowed")
+}
+
+func validate(obj *appsv1alpha1.EphemeralJob) error {
+
+	return nil
```
Comment (furykerry) [inline pkg/webhook/ephemeraljob/validating/ephemeraljob_create_update_handler.go:61]: include some logics from validateEphemeralContainers in k8s.io/kubernetes/pkg/apis/core/validation/validation.go

File: pkg/webhook/ephemeraljob/validating/validation.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child(lcName), errorText))
+		}
+	}
+
+	return allErrs
+}
+
+// validateEphemeralContainers is called by pod spec and template validation to validate the list of ephemeral containers.
+// Note that this is called for pod template even though ephemeral containers aren't allowed in pod templates.
+func validateEphemeralContainers(ephemeralContainers []core.EphemeralContainer, containers, initContainers []core.Container, volumes map[string]core.VolumeSource, podClaimNames sets.String, fldPath *field.Path, opts validation.PodValidationOptions) field.ErrorList {
```
Comment (furykerry) [inline pkg/webhook/ephemeraljob/validating/validation.go:208]: is it necessary to keep  containers, initContainers, volumes params if nil is always pass to this func

Comment (ABNER-1) [reply pkg/webhook/ephemeraljob/validating/validation.go]: I will remove parameters from the function that cannot be validated by the eJob validating webhook.

File: pkg/webhook/add_ephemeraljob.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2021 The Kruise Authors.
```
Comment (zmberg) [inline pkg/webhook/add_ephemeraljob.go:2]: 2024

File: pkg/webhook/ephemeraljob/validating/validation.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+		break
+	case "":
+		allErrors = append(allErrors, field.Required(fldPath, ""))
+	default:
+		allErrors = append(allErrors, field.NotSupported(fldPath, policy, supportedPullPolicies.List()))
+	}
+	return allErrors
+}
+
+func validateResizePolicy(policyList []core.ContainerResizePolicy, fldPath *field.Path) field.ErrorList {
```
Comment (zmberg) [inline pkg/webhook/ephemeraljob/validating/validation.go:99]: Is it still necessary to resize the temporary container?

Comment (ABNER-1) [reply pkg/webhook/ephemeraljob/validating/validation.go]: you're right.

**Pull request merged**